example/firmware-ariel-os/src/main.rs

```diff
@@ -0,0 +1,132 @@
+
+pub struct Context;
+
+type AppMutex = embassy_sync::blocking_mutex::raw::CriticalSectionRawMutex;
+type AppDriver = usb::UsbDriver;
+type AppStorage = WireStorageNoUsb<AppMutex, AppDriver>;
+type BufStorage = PacketBuffers<1024, 1024>;
+type AppTx = WireTxImpl<AppMutex, AppDriver>;
+type AppRx = WireRxImpl<AppDriver>;
+type AppServer = Server<AppTx, AppRx, WireRxBuf, MyApp>;
+
+static PBUFS: ConstStaticCell<BufStorage> = ConstStaticCell::new(BufStorage::new());
+static STORAGE: AppStorage = AppStorage::new();
+
+/// Helper to get unique ID from flash
+pub fn get_unique_id() -> Option<u64> {
+    // TODO
+    Some(12345678)
+}
+
+#[ariel_os::config(usb)]
+const USB_CONFIG: embassy_usb::Config = {
+    let mut config = embassy_usb::Config::new(0x16c0, 0x27DD);
+    config.manufacturer = Some("OneVariable");
+    config.product = Some("ov-twin");
+    config.serial_number = Some("12345678");
+
+    // Required for windows compatibility.
+    // https://developer.nordicsemi.com/nRF_Connect_SDK/doc/1.9.1/kconfig/CONFIG_CDC_ACM_IAD.html#help
+    config.device_class = 0xEF;
+    config.device_sub_class = 0x02;
+    config.device_protocol = 0x01;
+    config.composite_with_iads = true;
+
+    config
+};
+
+define_dispatch! {
+    app: MyApp;
+    spawn_fn: spawn_fn;
+    tx_impl: AppTx;
+    spawn_impl: WireSpawnImpl;
+    context: Context;
+
+    endpoints: {
+        list: ENDPOINT_LIST;
+
+        | EndpointTy                | kind      | handler                       |
+        | ----------                | ----      | -------                       |
+        | PingEndpoint              | blocking  | ping_handler                  |
+    };
+    topics_in: {
+        list: TOPICS_IN_LIST;
+
+        | TopicTy                   | kind      | handler                       |
+        | ----------                | ----      | -------                       |
+    };
+    topics_out: {
+        list: TOPICS_OUT_LIST;
+    };
+}
+
+#[ariel_os::task(autostart, usb_builder_hook)]
+async fn main() {
+    info!("Start");
+
+    let unique_id = get_unique_id().unwrap();
+    info!("id: {=u64:016X}", unique_id);
+
+    // USB/RPC INIT
+    let pbufs = PBUFS.take();
+    let context = Context;
+
+    // Create and inject the Postcard usb endpoint on the system USB builder.
+    let (tx_impl, rx_impl) = USB_BUILDER_HOOK
+        .with(|builder| STORAGE.init(builder, pbufs.tx_buf.as_mut_slice()))
+        .await;
+
+    let spawner = Spawner::for_current_executor().await;
+    let dispatcher = MyApp::new(context, spawner.into());
+    let vkk = dispatcher.min_key_len();
+    let mut server: AppServer = Server::new(
+        tx_impl,
+        rx_impl,
+        pbufs.rx_buf.as_mut_slice(),
+        dispatcher,
+        vkk,
+    );
+
+    loop {
+        // Somehow at least on nrf52840dk, this is needed, otherwise the
+        // `ariel_os_embassy::init_task()` task starves.
+        Timer::after(Duration::from_millis(100)).await;
```

Comment:
This stumped me. Somehow the other task (on the same executor) is starved without this delay here, and USB doesn't enumerate anymore.